vllm/v1/metrics/loggers.py

```diff
@@ -104,6 +106,12 @@ def __init__(self, model_config: ModelConfig):
             documentation="GPU KV-cache usage. 1 means 100 percent usage.",
             labelnames=labelnames).labels(*labelvalues)
 
+        self.gauge_gpu_prefix_cache_hit_rate = prometheus_client.Gauge(
```

Comment:
I know this was a Gauge in v0 too and every other metric we've added to v1, we've just re-used the v0 metric

But, in this case, I think it's worth considering adding `query_total` and `query_hit` Counters instead, and using Prometheus queries like this instead:

```rate(cache_query_hit[5m]) / rate(cache_query_total[5m])```

Consider how e.g. the hit rate drops due to a sudden increase in misses, that change will be diluted by the large accumulated totals.

You might be monitoring one set of instances that have just started up, and another set of instances that have been running for weeks ... if the hit rate drops consistently across all of those instances, it would only show up in the metrics of the recently started instances.

Reply:
Actually, the right thing to do would be to add the `hit_rate` Gauge with a "deprecated" note in the documentation string, and also add the two Counters :+1:

Reply:
That makes sense, but we should align the time interval (between two logs) of all metrics, so maybe we should just reset the metrics in the kv cache manager after `scheduler_stats.gpu_prefix_cache_hit_rate` is called. In this case we also don't have to deprecate `hit_rate`.

Reply:
With Prometheus, the time interval for the rate should be a user-defined thing - https://prometheus.io/docs/prometheus/latest/querying/functions/#rate - that's why you'd want the counters

With logging, yep I agree it would be good to reset the counters and calculate the hit rate over the interval between log outputs

I'm happy to take on figure this out if you like :+1:

Reply:
I think if we want to have the hit rate in LoggingStatLogger, we would need it to accumulate and reset its own counters

Reply:
Yeah that makes sense. If possible I still want to reset metrics in the scheduler so that the large number issue can be avoided at all. I'll try to figure out